csharp/src/Ice/Internal/Incoming.cs

```diff
@@ -309,22 +303,16 @@ public Task<Ice.OutputStream> setResultTask(Task task)
 
     public Task<Ice.OutputStream> setMarshaledResultTask<T>(Task<T> task) where T : Ice.MarshaledResult
     {
-        if (task == null)
-        {
-            return setResult(default(T).getOutputStream(_current));
-        }
-        else
+        Debug.Assert(task is not null);
```

Comment:
Same here. The user should not return a null task. It would be clearer with `#nullable enable`.

Reply:
They shouldn't but they can. We should still handle this situation and not crash.

Reply:
I think the previous code was worst as it hides the bug, now it will fail with `NullReferenceException` and the user can fix it.

Reply:
Right, I forgot the C# will throw an exception.